src/sentry/event_manager.py

```diff
@@ -389,9 +389,18 @@ def save(
         with metrics.timer("event_manager.load_grouping_config"):
             # At this point we want to normalize the in_app values in case the
             # clients did not set this appropriately so far.
-            grouping_config = get_grouping_config_dict_for_event_data(
-                job["event"].data.data, project
-            )
+            if is_reprocessed:
+                # The customer might have changed grouping enhancements since
+                # the event was ingested -> make sure we get the fresh one for reprocessing.
+                grouping_config = get_grouping_config_dict_for_project(project)
```

Comment:
Question: is there a chance of getting a cached version (and thus, not a fresh one) of the config?

Reply:
The config is fetched from project options, which does have a cache, but that one is updated when a new value is written, so it should always be fresh https://github.com/getsentry/sentry/blob/6b0b3d7657ff64eb0877d2f69e4031d5c139472b/src/sentry/models/options/project_option.py#L50